hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcConnection.java

```diff
@@ -149,6 +149,15 @@ private void scheduleRelogin(Throwable error) {
     if (error instanceof FallbackDisallowedException) {
       return;
     }
+    // do not schedule anything if not needed...
+    boolean isOverKrb = false;
```

Comment:
I do not think it is related to BufferedMutator? Please open a new issue if you want to optimize the kerberos stuff.

Reply:
I don't want to optimize the kerberos stuff, just avoid an used thread when kerberos is not used.